### PR TITLE
Add per-contig filters for mosdepth

### PR DESCRIPTION
**Pull request by nh13**
For genomes with many decoy, alts, random, and additional sequences, we may want to selectively remove contigs from the `mosdepth` per-contig plots.  I've added support for pattern matching, the same as the file inclusion/exclusion patterns.  The `mosdepth` docs have been updated.

- [x] This comment contains a description of changes (with reason)
- [x] `CHANGELOG.md` has been updated

### CODE REVIEW

**Comment by nh13** [comment]
@ewels all checks pass, ready for review!

File: docs/modules/mosdepth.md [hunk 1]
```diff
@@ -48,3 +48,21 @@ general_stats_coverage_hidden:
   - 20
   - 200
 ```
+
+For the per-contig coverage plot, you can include and exclude contigs based on name or pattern:
```
Comment (ewels) [inline docs/modules/mosdepth.md:52]:
```suggestion
For the per-contig coverage plot, you can include and exclude contigs based on name or pattern.

For example, you could add the following to your MultiQC config file:
```

File: multiqc/modules/mosdepth/mosdepth.py [hunk 2]
```diff
@@ -12,6 +12,8 @@
 from multiqc.modules.qualimap.QM_BamQC import coverage_histogram_helptext, genome_fraction_helptext
 from multiqc.plots import linegraph, bargraph
 
+import fnmatch
```
Comment (ewels) [inline multiqc/modules/mosdepth/mosdepth.py:15]: Nitpick: Imports are a bit messy here already, but anyway if you could move next to `import logging` that would be nice 👍🏻

File: multiqc/modules/mosdepth/mosdepth.py [hunk 3]
```diff
@@ -205,6 +221,20 @@ def parse_cov_dist(self):
                         if cumcov > 1:  # require >1% to prevent long flat tail
                             xmax = max(xmax, x)
                     else:  # for per-contig plot
+                        # filter out contigs based on exclusion patterns
+                        if any(fnmatch.fnmatch(contig, pattern) for pattern in exclude_contigs):
+                            # Commented out since this could be many thousands of contigs fo reach!
+                            # log.debug("Skipping excluded contig '{}'".format(contig))
+                            continue
```
Comment (ewels) [inline multiqc/modules/mosdepth/mosdepth.py:228]: What I have done for similar cases in the past is make these debug statements disabled by default, but have a config option to enable them. Can very occasionally be useful for debugging and not tricky to add.

```suggestion
                            if config.mosdepth_config.get('log_contig_exclusion'):
                                log.debug("Skipping excluded contig '{}'".format(contig))
```

**Review by ewels — COMMENTED** [review]
Code looks good, and nice addition - thanks! 👍🏻 

Tried running locally on the [mosdepth test data](https://github.com/ewels/MultiQC_TestData/tree/master/data/modules/mosdepth) but it crashed though:

```
  /// MultiQC 🔍 | v1.11.dev0 (aec7887)

|           multiqc | Search path : /Users/ewels/GitHub/MultiQC/TestData/data/modules/mosdepth
|         searching | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 19/19
|          mosdepth | Found 4 reports
╭───────────────────────────────────────────── Oops! The 'mosdepth' MultiQC module broke... ─────────────────────────────────────────────╮
│ Please copy this log and report it at https://github.com/ewels/MultiQC/issues                                                          │
│ Please attach a file that triggers the error. The last file found was: ./viralrecon/sample1.trim.mkD.amplicon.mosdepth.global.dist.txt │
│                                                                                                                                        │
│ Traceback (most recent call last):                                                                                                     │
│   File "/Users/ewels/GitHub/MultiQC/MultiQC/multiqc/multiqc.py", line 604, in run                                                      │
│     output = mod()                                                                                                                     │
│   File "/Users/ewels/GitHub/MultiQC/MultiQC/multiqc/modules/mosdepth/mosdepth.py", line 171, in __init__                               │
│     threshs, hidden_threshs = get_cov_thresholds()                                                                                     │
│   File "/Users/ewels/GitHub/MultiQC/MultiQC/multiqc/modules/mosdepth/mosdepth.py", line 293, in get_cov_thresholds                     │
│     threshs = config.mosdepth_config["general_stats_coverage"]                                                                         │
│ KeyError: 'general_stats_coverage'                                                                                                     │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
|           multiqc | No analysis results found. Cleaning up..
|           multiqc | MultiQC complete
```

If you could take a look at this that would be fab. Then happy to merge..

**Comment by ewels** [comment]
> Tried running locally on the [mosdepth test data](https://github.com/ewels/MultiQC_TestData/tree/master/data/modules/mosdepth) but it crashed though:

Coming back to look at this - it's because the existing code couldn't handle having a mosdepth config that didn't have the keys it was expecting. Adding an additional exception type to catch resolves this.

**Comment by ewels** [comment]
It looks like PR https://github.com/ewels/MultiQC/pull/1437 scooped these commits in as well. To keep things simple I am pushing tweaks + fixes to this code into that PR, and I will merge it all in one go.

**Pull request merged**